File: julia/get-files.jl
indentationWidth = 2;
ignoredPaths = [".", "..", ".git", "node_modules", "target", "love", ".dub", "dist"];

function doFiles(directoryPath = pwd(), traversalLevel = 0, callback = println)
    indentation = ' '^(indentationWidth * traversalLevel)
    for entry in readdir(directoryPath)
        callback(indentation * "$entry")
        fullPath = abspath("$directoryPath/$entry")
        if !isdir(fullPath) || in(entry, ignoredPaths) continue end
        doFiles(fullPath, traversalLevel + 1, callback)
    end
end

length(ARGS) > 0 ? doFiles(popfirst!(ARGS)) : doFiles(pwd())

# Local Variables:
# compile-command: "julia ./get-files.jl .."
# End:

File: python/get-files.py
from os import scandir

INDENTATION_WIDTH = 2
IGNORED_PATHS = [".git", "node_modules", "target", "dist", ".dub", "love"]

def doFiles(directoryPath = ".", traversalLevel = 0, callback = print):
    for entry in scandir(directoryPath):
        callback(' ' * (INDENTATION_WIDTH * traversalLevel) + entry.name)
        if entry.is_dir() and not entry.name in IGNORED_PATHS:
            doFiles(directoryPath + "/" + entry.name, traversalLevel + 1, callback)

printFiles = doFiles("..", 0, print)

printFiles()

# Local Variables:
# compile-command: "python3 ./get-files.py"
# End:
